src/pipeline/pipeline_construction.c

```diff
@@ -155,7 +185,14 @@ static ResultProcessor *getArrangeRP(Pipeline *pipeline, const AggregationPipeli
 
   size_t maxResults = astp->offset + astp->limit;
   if (!maxResults) {
-    maxResults = DEFAULT_LIMIT;
+    if (HasWithCount(&params->common)) {
+      // No LIMIT specified, consume up to maxResultsLimit
+      maxResults = params->maxResultsLimit;
```

Comment:
Important: Is this affecting the actual rows being returned? (from shard to the coord)
Need to only set the total count, without adding more rows to the response.